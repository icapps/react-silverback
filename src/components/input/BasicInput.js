import React from 'react';
import PropTypes from 'prop-types';

class TextInput extends React.Component {
  handleKeyPress = event => {
    if (event.key === 'Enter') {
      this.props.handleEnter();
    }
  }

  render() {
    const { props } = this;
    return (
      <div className="form-group">
        <label htmlFor={props.id}>{props.label}</label>
        <input type={props.type === 'email' ? 'text': props.type} className={`form-control ${!props.isValid ? 'is-invalid' : ''}`} id={props.id} placeholder={props.placeholder} value={props.value} onKeyPress={this.handleKeyPress} onChange={props.handleChange} disabled={props.isDisabled} />
        {!props.isValid && <div className="invalid-feedback">{props.errorMessage}</div>}
      </div>
    );
  }
}


TextInput.propTypes = {
  id: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  handleChange: PropTypes.func.isRequired,
  type: PropTypes.oneOf(['email', 'text', 'password', 'number']),
  isValid: PropTypes.bool,
  errorMessage: PropTypes.string,
  isDisabled: PropTypes.bool,
  handleEnter: PropTypes.func,
};

TextInput.defaultProps = {
  placeholder: '',
  type: 'text',
  isValid: true,
  errorMessage: '',
  isDisabled: false,
  handleEnter: () => { },
};

export default TextInput;
