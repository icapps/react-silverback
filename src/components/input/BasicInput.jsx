import React from 'react';
import PropTypes from 'prop-types';

const TextInput = props => (
  <div className="form-group">
    <label htmlFor={props.id}>{props.label}</label>
    <input type={props.type} className={`form-control ${!props.isValid ? 'is-invalid' : ''}`} id={props.id} placeholder={props.placeholder} value={props.value} onChange={props.handleChange} />
    {!props.isValid && <div class="invalid-feedback">{props.errorMessage}</div>}
  </div>
);

TextInput.propTypes = {
  id: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  handleChange: PropTypes.func.isRequired,
  type: PropTypes.oneOf(['text', 'password', 'number']),
  isValid: PropTypes.bool,
  errorMessage: PropTypes.string,
};

TextInput.defaultProps = {
  placeholder: '',
  type: 'text',
  isValid: true,
  errorMessage: '',
};

export default TextInput;
