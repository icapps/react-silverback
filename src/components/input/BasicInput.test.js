import React from 'react';
import { shallow, configure } from 'enzyme';
import BasicInput from './BasicInput';
import Adapter from 'enzyme-adapter-react-16';


configure({ adapter: new Adapter() });

describe('BasicInput Component', () => {
  it('should render a BasicInput component', () => {
    const wrapper = shallow(<BasicInput id="test" label="Text" value="test" handleChange={() => { }} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a BasicInput component with a placeholder', () => {
    const wrapper = shallow(<BasicInput id="test" label="Text with placeholder" placeholder="placeholder" value="" handleChange={() => { }} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a BasicInput component with a type of password', () => {
    const wrapper = shallow(<BasicInput id="test" label="Password" value="password" handleChange={() => { }} type="password" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a BasicInput component with a type of number', () => {
    const wrapper = shallow(<BasicInput id="test" label="Number" value={123} handleChange={() => { }} type="number" />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render a BasicInput component with error handling', () => {
    const wrapper = shallow(<BasicInput id="test" label="Text with errorhandling" value="test" handleChange={() => { }} isValid={false} errorMessage="This is an error message" />);
    expect(wrapper).toMatchSnapshot();
  });
});
