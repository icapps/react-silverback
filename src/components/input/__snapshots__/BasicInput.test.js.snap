// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicInput Component should render a BasicInput component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextInput
    errorMessage=""
    handleChange={[Function]}
    handleEnter={[Function]}
    id="test"
    isDisabled={false}
    isValid={true}
    label="Text"
    placeholder=""
    type="text"
    value="test"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          htmlFor="test"
>
          Text
</label>,
        <input
          className="form-control "
          disabled={false}
          id="test"
          onChange={[Function]}
          onKeyPress={[Function]}
          placeholder=""
          type="text"
          value="test"
/>,
        false,
      ],
      "className": "form-group",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Text",
          "htmlFor": "test",
        },
        "ref": null,
        "rendered": "Text",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "form-control ",
          "disabled": false,
          "id": "test",
          "onChange": [Function],
          "onKeyPress": [Function],
          "placeholder": "",
          "type": "text",
          "value": "test",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            htmlFor="test"
>
            Text
</label>,
          <input
            className="form-control "
            disabled={false}
            id="test"
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder=""
            type="text"
            value="test"
/>,
          false,
        ],
        "className": "form-group",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Text",
            "htmlFor": "test",
          },
          "ref": null,
          "rendered": "Text",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control ",
            "disabled": false,
            "id": "test",
            "onChange": [Function],
            "onKeyPress": [Function],
            "placeholder": "",
            "type": "text",
            "value": "test",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`BasicInput Component should render a BasicInput component that can handle an enter 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextInput
    errorMessage="This is an error message"
    handleChange={[Function]}
    handleEnter={[Function]}
    id="test"
    isDisabled={false}
    isValid={false}
    label="Text with errorhandling"
    placeholder=""
    type="text"
    value="test"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          htmlFor="test"
>
          Text with errorhandling
</label>,
        <input
          className="form-control is-invalid"
          disabled={false}
          id="test"
          onChange={[Function]}
          onKeyPress={[Function]}
          placeholder=""
          type="text"
          value="test"
/>,
        <div
          className="invalid-feedback"
>
          This is an error message
</div>,
      ],
      "className": "form-group",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Text with errorhandling",
          "htmlFor": "test",
        },
        "ref": null,
        "rendered": "Text with errorhandling",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "form-control is-invalid",
          "disabled": false,
          "id": "test",
          "onChange": [Function],
          "onKeyPress": [Function],
          "placeholder": "",
          "type": "text",
          "value": "test",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "This is an error message",
          "className": "invalid-feedback",
        },
        "ref": null,
        "rendered": "This is an error message",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            htmlFor="test"
>
            Text with errorhandling
</label>,
          <input
            className="form-control is-invalid"
            disabled={false}
            id="test"
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder=""
            type="text"
            value="test"
/>,
          <div
            className="invalid-feedback"
>
            This is an error message
</div>,
        ],
        "className": "form-group",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Text with errorhandling",
            "htmlFor": "test",
          },
          "ref": null,
          "rendered": "Text with errorhandling",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control is-invalid",
            "disabled": false,
            "id": "test",
            "onChange": [Function],
            "onKeyPress": [Function],
            "placeholder": "",
            "type": "text",
            "value": "test",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "This is an error message",
            "className": "invalid-feedback",
          },
          "ref": null,
          "rendered": "This is an error message",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`BasicInput Component should render a BasicInput component with a placeholder 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextInput
    errorMessage=""
    handleChange={[Function]}
    handleEnter={[Function]}
    id="test"
    isDisabled={false}
    isValid={true}
    label="Text with placeholder"
    placeholder="placeholder"
    type="text"
    value=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          htmlFor="test"
>
          Text with placeholder
</label>,
        <input
          className="form-control "
          disabled={false}
          id="test"
          onChange={[Function]}
          onKeyPress={[Function]}
          placeholder="placeholder"
          type="text"
          value=""
/>,
        false,
      ],
      "className": "form-group",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Text with placeholder",
          "htmlFor": "test",
        },
        "ref": null,
        "rendered": "Text with placeholder",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "form-control ",
          "disabled": false,
          "id": "test",
          "onChange": [Function],
          "onKeyPress": [Function],
          "placeholder": "placeholder",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            htmlFor="test"
>
            Text with placeholder
</label>,
          <input
            className="form-control "
            disabled={false}
            id="test"
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder="placeholder"
            type="text"
            value=""
/>,
          false,
        ],
        "className": "form-group",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Text with placeholder",
            "htmlFor": "test",
          },
          "ref": null,
          "rendered": "Text with placeholder",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control ",
            "disabled": false,
            "id": "test",
            "onChange": [Function],
            "onKeyPress": [Function],
            "placeholder": "placeholder",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`BasicInput Component should render a BasicInput component with a type of number 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextInput
    errorMessage=""
    handleChange={[Function]}
    handleEnter={[Function]}
    id="test"
    isDisabled={false}
    isValid={true}
    label="Number"
    placeholder=""
    type="number"
    value={123}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          htmlFor="test"
>
          Number
</label>,
        <input
          className="form-control "
          disabled={false}
          id="test"
          onChange={[Function]}
          onKeyPress={[Function]}
          placeholder=""
          type="number"
          value={123}
/>,
        false,
      ],
      "className": "form-group",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Number",
          "htmlFor": "test",
        },
        "ref": null,
        "rendered": "Number",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "form-control ",
          "disabled": false,
          "id": "test",
          "onChange": [Function],
          "onKeyPress": [Function],
          "placeholder": "",
          "type": "number",
          "value": 123,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            htmlFor="test"
>
            Number
</label>,
          <input
            className="form-control "
            disabled={false}
            id="test"
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder=""
            type="number"
            value={123}
/>,
          false,
        ],
        "className": "form-group",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Number",
            "htmlFor": "test",
          },
          "ref": null,
          "rendered": "Number",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control ",
            "disabled": false,
            "id": "test",
            "onChange": [Function],
            "onKeyPress": [Function],
            "placeholder": "",
            "type": "number",
            "value": 123,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`BasicInput Component should render a BasicInput component with a type of password 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextInput
    errorMessage=""
    handleChange={[Function]}
    handleEnter={[Function]}
    id="test"
    isDisabled={false}
    isValid={true}
    label="Password"
    placeholder=""
    type="password"
    value="password"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          htmlFor="test"
>
          Password
</label>,
        <input
          className="form-control "
          disabled={false}
          id="test"
          onChange={[Function]}
          onKeyPress={[Function]}
          placeholder=""
          type="password"
          value="password"
/>,
        false,
      ],
      "className": "form-group",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Password",
          "htmlFor": "test",
        },
        "ref": null,
        "rendered": "Password",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "form-control ",
          "disabled": false,
          "id": "test",
          "onChange": [Function],
          "onKeyPress": [Function],
          "placeholder": "",
          "type": "password",
          "value": "password",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            htmlFor="test"
>
            Password
</label>,
          <input
            className="form-control "
            disabled={false}
            id="test"
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder=""
            type="password"
            value="password"
/>,
          false,
        ],
        "className": "form-group",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Password",
            "htmlFor": "test",
          },
          "ref": null,
          "rendered": "Password",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control ",
            "disabled": false,
            "id": "test",
            "onChange": [Function],
            "onKeyPress": [Function],
            "placeholder": "",
            "type": "password",
            "value": "password",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`BasicInput Component should render a BasicInput component with error handling 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextInput
    errorMessage="This is an error message"
    handleChange={[Function]}
    handleEnter={[Function]}
    id="test"
    isDisabled={false}
    isValid={false}
    label="Text with errorhandling"
    placeholder=""
    type="text"
    value="test"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          htmlFor="test"
>
          Text with errorhandling
</label>,
        <input
          className="form-control is-invalid"
          disabled={false}
          id="test"
          onChange={[Function]}
          onKeyPress={[Function]}
          placeholder=""
          type="text"
          value="test"
/>,
        <div
          className="invalid-feedback"
>
          This is an error message
</div>,
      ],
      "className": "form-group",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Text with errorhandling",
          "htmlFor": "test",
        },
        "ref": null,
        "rendered": "Text with errorhandling",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "form-control is-invalid",
          "disabled": false,
          "id": "test",
          "onChange": [Function],
          "onKeyPress": [Function],
          "placeholder": "",
          "type": "text",
          "value": "test",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "This is an error message",
          "className": "invalid-feedback",
        },
        "ref": null,
        "rendered": "This is an error message",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            htmlFor="test"
>
            Text with errorhandling
</label>,
          <input
            className="form-control is-invalid"
            disabled={false}
            id="test"
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder=""
            type="text"
            value="test"
/>,
          <div
            className="invalid-feedback"
>
            This is an error message
</div>,
        ],
        "className": "form-group",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Text with errorhandling",
            "htmlFor": "test",
          },
          "ref": null,
          "rendered": "Text with errorhandling",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control is-invalid",
            "disabled": false,
            "id": "test",
            "onChange": [Function],
            "onKeyPress": [Function],
            "placeholder": "",
            "type": "text",
            "value": "test",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "This is an error message",
            "className": "invalid-feedback",
          },
          "ref": null,
          "rendered": "This is an error message",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
